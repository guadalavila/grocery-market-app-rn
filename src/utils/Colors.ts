export const Colors = {
    primary: '#68B984',
    primaryLight: '#CFFDE1',
    greenUltraLight: '#E9F9F2',
    green: '#2DA32A',
    secondary: '#3D5656',
    yellow: '#FED049',
    orange: '#FCB753',
    white: '#FFFFFF',
    black: '#000000',
    greyLight: '#C5C5C5',
    greyUltraLight: '#F7F8FA',
    background: '#FCFDFB',
    grey: '#9f9f9f',
    divider: '#EFEEF1',
};
