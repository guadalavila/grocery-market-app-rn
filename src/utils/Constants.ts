export const STORAGE_KEY = {
    FAVORITE: '@favorite',
};
