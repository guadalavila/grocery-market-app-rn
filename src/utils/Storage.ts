import AsyncStorage from '@react-native-async-storage/async-storage';

export const storeData = async (key: string, value: string) => {
    try {
        const jsonValue = JSON.stringify(value);
        await AsyncStorage.setItem(key, jsonValue);
    } catch (e) {}
};

export const getData = async (key: string) => {
    try {
        const data = await AsyncStorage.getItem(key);
        return data ? JSON.parse(data) : undefined;
    } catch (e) {
        // error reading value
    }
};
