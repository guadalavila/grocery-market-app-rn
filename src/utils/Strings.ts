export const Strings = {
    appName: 'Grocery Market',
    hello: 'Hola!',
    welcome: 'Bienvenido!',
    viewMore: 'Ver MÃ¡s',
    search: 'Buscar...',

    cart: {
        label: 'Carrito',
        empty: 'No tenes productos en el carrito',
        checkout: 'Terminar Compra',
        add: 'Agregar al carrito',
    },

    addedProduct: 'Producto Agregado!',

    fruits: {
        label: 'Frutas',
    },
    vegetables: {
        label: 'Verduras',
    },

    favorites: {
        label: 'Mis Favoritos',
        empty: 'No tenes productos como favoritos',
    },
};
