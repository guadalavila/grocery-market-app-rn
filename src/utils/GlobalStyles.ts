import { StyleSheet } from 'react-native';
import { Colors } from './Colors';

export const GlobalStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    shadowCard: {
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 4,
    },
    row: {
        flexDirection: 'row',
    },
    justBtw: {
        justifyContent: 'space-between',
    },
    main: {
        backgroundColor: Colors.background,
    },
});

export const flex = (x: number) => {
    return {
        flex: x,
    };
};

export const Sizes = {
    XXS: 2,
    XS: 4,
    S: 8,
    M: 16,
    L: 24,
    XL: 30,
    XXL: 40,
    XXXL: 55,
};

export const FontSize = {
    XS: 8,
    S: 10,
    M: 14,
    L: 18,
    XL: 22,
    XXL: 26,
    XXXL: 32,
};
