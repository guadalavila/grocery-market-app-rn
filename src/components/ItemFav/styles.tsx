import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        backgroundColor: Colors.white,
        borderRadius: Sizes.S,
        paddingHorizontal: Sizes.L,
        padding: Sizes.M,
        marginVertical: Sizes.S,
        marginHorizontal: Sizes.L,
        flexDirection: 'row',
    },
    image: {
        width: 100,
        height: 70,
    },
    title: {
        fontWeight: '900',
        fontSize: 16,
        color: Colors.black,
        alignSelf: 'center',
    },
    button: {
        alignSelf: 'center',
        backgroundColor: Colors.redLight,
        borderRadius: 20,
        padding: 10,
    },
});
