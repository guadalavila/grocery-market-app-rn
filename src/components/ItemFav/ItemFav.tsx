import React, { useContext } from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Product } from '../../models/Product';
import { flex, GlobalStyles } from '../../utils/GlobalStyles';
import { styles } from './styles';
import { Colors } from '../../utils/Colors';
import { FavsContext } from '../../context/favs-context';

interface IItemFavProps {
    product: Product;
}
const ItemFav = ({ product }: IItemFavProps) => {
    const { addRemoveProduct } = useContext(FavsContext);
    return (
        <View style={styles.container}>
            <View style={[GlobalStyles.row, flex(1), GlobalStyles.justBtw]}>
                <Image testID='image' resizeMode='center' style={styles.image} source={{ uri: product.image }} />
                <Text style={styles.title}>{product.name}</Text>
                <TouchableOpacity
                    testID='btn-fav'
                    activeOpacity={0.7}
                    onPress={() => addRemoveProduct(product)}
                    style={styles.button}>
                    <Icon testID='icon-heart' size={20} name='heart' color={Colors.red} />
                </TouchableOpacity>
            </View>
        </View>
    );
};

export default ItemFav;
