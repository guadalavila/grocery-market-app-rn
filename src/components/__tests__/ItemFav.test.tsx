import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import { Colors } from '../../utils/Colors';
import ItemFav from '../ItemFav/ItemFav';

describe('test ItemFav component', () => {
    const productMock = {
        id: '6b17d300-943a-11ed-a1eb-0242ac120002',
        image: 'https://res.cloudinary.com/deoaxotzs/image/upload/v1673720991/grocery-store/banana-_yagtxh.webp',
        name: 'Banana',
        description: 'Organicas',
        price: 20.0,
    };

    test('should match with snapshot', () => {
        const component = render(<ItemFav product={productMock} />);
        expect(component).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<ItemFav product={productMock} />);
        expect(screen).toBeTruthy();
    });
    test('should render text', () => {
        render(<ItemFav product={productMock} />);
        expect(screen.getByText(productMock.name)).toBeTruthy();
    });

    test('should render image', () => {
        render(<ItemFav product={productMock} />);
        expect(screen.getByTestId('image')).toBeTruthy();
        expect(screen.getByTestId('image').props.resizeMode).toBe('center');
    });

    test('should render icon', () => {
        render(<ItemFav product={productMock} />);
        expect(screen.getByTestId('icon-heart')).toBeTruthy();
        expect(screen.getByTestId('icon-heart').props.name).toBe('heart');
        expect(screen.getByTestId('icon-heart').props.color).toBe(Colors.red);
        expect(screen.getByTestId('icon-heart').props.size).toBe(20);
    });

    test('should call fn', () => {
        render(<ItemFav product={productMock} />);
        const button = screen.getByTestId('btn-fav');
        expect(button).toBeTruthy();
        fireEvent.press(button);
    });
});
