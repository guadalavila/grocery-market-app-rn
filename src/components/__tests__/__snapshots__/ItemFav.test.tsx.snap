// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemFav component should match with snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "flexDirection": "row",
      "marginHorizontal": 24,
      "marginVertical": 8,
      "padding": 16,
      "paddingHorizontal": 24,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Image
      resizeMode="center"
      source={
        Object {
          "uri": "https://res.cloudinary.com/deoaxotzs/image/upload/v1673720991/grocery-store/banana-_yagtxh.webp",
        }
      }
      style={
        Object {
          "height": 70,
          "width": 100,
        }
      }
      testID="image"
    />
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "900",
        }
      }
    >
      Banana
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFE8E8",
          "borderRadius": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="btn-fav"
    >
      <MockedIoniconsIcons
        color="#FF1C1E"
        name="heart"
        size={20}
        testID="icon-heart"
      />
    </View>
  </View>
</View>
`;
